### PR TITLE
AdSecGH-65 Business Logic Separation

### PR DESCRIPTION
**Pull request by psarras**

### CODE REVIEW

File: AdSecGHTests/Parameters/AdSecSectionTests.cs [hunk 1]
```diff
@@ -72,27 +78,37 @@ private void TestFlattenedSection(ISection expected, ISection actual) {
...
-      Assert.Equal(expectedProfile.ElasticModulus().Z.Value, actualProfile.ElasticModulus().Z.Value, new DoubleComparer());
-      Assert.Equal(expectedProfile.SurfaceAreaPerUnitLength().Value, actualProfile.SurfaceAreaPerUnitLength().Value, new DoubleComparer());
-      Assert.Equal(expectedProfile.LocalAxisSecondMomentOfArea().YY.Value, actualProfile.LocalAxisSecondMomentOfArea().YY.Value, new DoubleComparer());
-      Assert.Equal(expectedProfile.LocalAxisSecondMomentOfArea().ZZ.Value, actualProfile.LocalAxisSecondMomentOfArea().ZZ.Value, new DoubleComparer());
-      Assert.Equal(expectedProfile.LocalAxisSecondMomentOfArea().YZ.Value, actualProfile.LocalAxisSecondMomentOfArea().YZ.Value, new DoubleComparer());
-      Assert.Equal(expectedProfile.PrincipalAxisSecondMomentOfArea().UU.Value, actualProfile.PrincipalAxisSecondMomentOfArea().UU.Value, new DoubleComparer());
-      Assert.Equal(expectedProfile.PrincipalAxisSecondMomentOfArea().VV.Value, actualProfile.PrincipalAxisSecondMomentOfArea().VV.Value, new DoubleComparer());
+      Assert.Equal(expectedProfile.ElasticModulus().Y.Value, actualProfile.ElasticModulus().Y.Value,
```
[the remark this answers is not in the document]
Comment (psarras) [reply AdSecGHTests/Parameters/AdSecSectionTests.cs:96]: sure but this test, is not part of the PR. Besides minimal changes, like removing unused variables, code, I am avoiding doing drastic changes on classes we are not focusing on. This PR was about separating the Business Logic on a separate project, some extra things were edited, to make sure the namespaces are getting loaded properly, and subsequently this triggered the auto-format, which made the changes appear more that they are.

File: AdSecGH/Helpers/Result.cs [hunk 2]
```diff
@@ -1,94 +1,94 @@
...
-        double tempmin = scale * Math.Round(min / scale, 4);
-        tempmin = Math.Floor(tempmin * 1000) / 1000;
-        roundedvals.Add(tempmin);
-      }
-
-      return roundedvals;
-    }
-  }
-}
+﻿// using System;
```
[the remark this answers is not in the document]
Comment (psarras) [reply AdSecGH/Helpers/Result.cs:1]: you are correct, have done this one the next PR

File: AdSecGHTests/Helpers/Duplicates.cs [hunk 3]
```diff
@@ -4,36 +4,118 @@
...
+    public static GH_Component[] CreateInstancesWithCreateInterface() {
+      var allInterfaceTypes = GetAllInterfaceTypes();
+      var components = allInterfaceTypes.Select(Activator.CreateInstance).Select(x => x as GH_Component).ToArray();
+      return components;
+    }
+  }
+
+  [Collection("GrasshopperFixture collection")]
+  public class DropToCanvasTests {
+    //[Fact]
```
[the remark this answers is not in the document]
Comment (psarras) [reply AdSecGHTests/Helpers/Duplicates.cs:63]: has been corrected on the next PR #100

File: AdSecGHTests/Parameters/AdSecSectionTests.cs [hunk 4]
```diff
@@ -1,42 +1,48 @@
...
-      ISection flattened = adSec.Flatten(section);
+      var flattened = adSec.Flatten(section);
 
       string fileName = Path.GetTempPath() + "AdSecSectionTest.ads";
       File.WriteAllText(fileName, CreateJson(designCode, section));
 
       string json = File.ReadAllText(fileName);
-      ParsedResult jsonParser = JsonParser.Deserialize(json);
-      ISection actualSection = jsonParser.Sections[0];
+      var jsonParser = JsonParser.Deserialize(json);
```
[the remark this answers is not in the document]
Comment (psarras) [reply AdSecGHTests/Parameters/AdSecSectionTests.cs:44]: good point, will pick it up on #100

**Review by DominikaLos — APPROVED** [review]

**Review by SandeepArup — APPROVED** [review]

**Pull request merged**